src/martian/directive.py

```diff
@@ -147,7 +147,10 @@ def _default(mro, get_default):
     # if we haven't found a result, raise the first error we had as
     # a GrokError
     if error is not None:
-        raise GrokError(unicode(error), error.component)
+        if sys.version_info[0] < 3:
+            raise GrokError(unicode(error), error.component)
+        else:
+            raise GrokError(str(error), error.component)
```

Comment:
I'd rather avoid an if here and instead have something like [six.text_type](https://pythonhosted.org/six/#six.text_type) that' an alias for unicode on Python 2 and str on Python 3.  You could define it in compat3.py, if you don't want to introduce a dependency on six.